### ISSUE TITLE
Compatibility with ES 2.4x

### ISSUE THREAD

**Issue by diclophis** [uken/fluent-plugin-elasticsearch#379]
#### Problem

Unable to send logs to ES via plugin due to following error message:

```
2018-03-07 06:37:40 +0000 [warn]: #0 [elasticsearch-output] Elasticsearch rejected 1 documents due to invalid field arguments
2018-03-07 06:37:40 +0000 [error]: #0 [elasticsearch-output] Could not push log to Elasticsearch: {"took"=>5, "errors"=>true, "items"=>[{"create"=>{"_index"=>"logstash-2018.03.07", "_type"=>"_doc", "_id"=>"AWH_LgJfGG_ML6qe9U3s", "status"=>400, "error"=>{"type"=>"invalid_type_name_exception", "reason"=>"Document mapping type name can't start with '_'"}}}]}
```

#### Steps to replicate

Our configuration was working with `td-agent 2x (0.12.x)` and `"fluent-plugin-elasticsearch v1.9.5` and Elasticsearch `2.4.x`

#### Expected Behavior or What you need to ask

Possibly support ES 2.4?

#### Using Fluentd and ES plugin versions

```
td-agent --version
td-agent 1.0.2
```

```
# cat /etc/issue
Ubuntu 14.04.5 LTS \n \l
```

```
# dpkg -l | grep td-agent
ii  td-agent                         3.1.1-0                                    amd64        Treasure Agent: A data collector for Treasure Data
```

```
# td-agent-gem list

*** LOCAL GEMS ***

addressable (2.5.2)
aws-partitions (1.42.0)
aws-sdk (2.11.9)
aws-sdk-core (3.11.0, 2.11.9)
aws-sdk-ec2 (1.29.0)
aws-sdk-kms (1.3.0)
aws-sdk-resources (2.11.9)
aws-sdk-s3 (1.8.0)
aws-sdk-sqs (1.3.0)
aws-sigv4 (1.0.2)
bigdecimal (default: 1.3.0)
bundler (1.16.0)
cool.io (1.5.3)
did_you_mean (1.1.0)
dig_rb (1.0.1)
elasticsearch (5.0.4)
elasticsearch-api (5.0.4)
elasticsearch-transport (5.0.4)
excon (0.60.0)
faraday (0.13.1)
ffi (1.9.23)
fluent-logger (0.7.1)
fluent-mixin-rewrite-tag-name (0.1.0)
fluent-plugin-elasticsearch (2.8.0, 2.4.0, 1.9.5)
fluent-plugin-elb-log (0.5.0)
fluent-plugin-geoip (1.2.0, 0.7.0)
fluent-plugin-kafka (0.6.5)
fluent-plugin-mysqlslowquery (0.0.8, 0.0.7)
fluent-plugin-parser (0.6.1)
fluent-plugin-rewrite-tag-filter (2.0.1)
fluent-plugin-s3 (1.1.1, 1.1.0, 0.8.3)
fluent-plugin-systemd (0.3.1, 0.0.8)
fluent-plugin-td (1.0.0)
fluent-plugin-td-monitoring (0.2.3)
fluent-plugin-webhdfs (1.2.2)
fluentd (1.0.2, 0.12.42)
geoip-c (0.9.1)
geoip2_c (0.3.2)
hirb (0.7.3)
http_parser.rb (0.6.0)
httpclient (2.8.2.4)
io-console (default: 0.4.6)
ipaddress (0.8.3)
jmespath (1.3.1)
json (default: 2.0.4)
ltsv (0.1.0)
mini_portile2 (2.3.0)
minitest (5.10.1)
mixlib-cli (1.7.0)
mixlib-config (2.2.4)
mixlib-log (1.7.1)
mixlib-shellout (2.2.7)
msgpack (1.2.0)
multi_json (1.12.2)
multipart-post (2.0.0)
myslog (0.1.1)
net-telnet (0.1.1)
nokogiri (1.8.1)
ohai (6.20.0)
oj (3.3.9)
openssl (default: 2.0.5)
parallel (1.12.1)
power_assert (0.4.1)
psych (default: 2.2.2)
public_suffix (3.0.1)
rake (12.0.0)
rdoc (default: 5.0.0)
ruby-kafka (0.5.1)
ruby-progressbar (1.9.0)
rubyzip (1.2.1)
serverengine (2.0.5)
sigdump (0.2.4)
string-scrub (0.0.5)
strptime (0.1.9)
systemd-journal (1.3.1)
systemu (2.5.2)
td (0.15.7)
td-client (1.0.4)
td-logger (0.3.27)
test-unit (3.2.3)
thread_safe (0.3.6)
tzinfo (1.2.4)
tzinfo-data (1.2017.3)
webhdfs (0.8.0)
xmlrpc (0.2.1)
yajl-ruby (1.3.1)
zip-zip (0.3)
```

**Comment by cosmo0920** [comment]
> Our configuration was working with td-agent 2x (0.12.x) and "fluent-plugin-elasticsearch v1.9.5 and Elasticsearch 2.4.x

Which version did you use elasticsearch gem with previous configuration?
Currently, it seems that you use elasticsearch gem v5.0.4 that is compatible for ES 5.x.
ref: https://github.com/elastic/elasticsearch-ruby#compatibility

**Comment by cosmo0920** [comment]
Ah! fluent-plugin-elasticsearch v2.8.0 contains misunderstood default value of type_name parameter.
This default value is compatible for ES 6.2.x or later.
This issue is solved in v2.8.2.
If you hit this issue again after upgrade to ES v2.8.2 or later, please add comment in this issue.
Anyway, thank you for your report!
